packages/agents/src/functions/DevelopmentPlanner.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  Agent,
+  AgentFunctionResult,
+  LlmApi,
+  ChatLogs,
+  Tokenizer,
+  ChatMessageBuilder,
+  ChatMessage,
+} from "@evo-ninja/agent-utils";
+import { AgentFunctionBase } from "../AgentFunctionBase";
+import { AgentBaseContext } from "../AgentBase";
+
+interface FunctionParams {
+  goal: string;
+  context?: string;
+  summarizedInfo?: string;
+}
+
+export class DevelopmentPlanner extends AgentFunctionBase<FunctionParams> {
+  constructor(
+    private _llm: LlmApi,
+    private _tokenizer: Tokenizer
+  ) {
+    super();
+  }
+
+  get name() {
+    return "planDevelopmentTasks";
+  }
+
+  get description() {
+    return `Create execution steps to tackle any software development task`;
+  }
+
+  get parameters() {
+    return {
+      type: "object",
+      properties: {
+        goal: {
+          type: "string",
+        },
+        context: {
+          type: "string",
+        },
+        summarizedInfo: {
```

Comment:
how does this differ from "context"? I think it's helpful to add a `description` field to properties so the caller (llm) knows what each one represents.